crawlerr/services/rag_service.py

```
@@ -73,6 +118,201 @@ def chunk_text(self, text: str, metadata: Dict[str, Any] = None) -> List[Dict[st
         if not text.strip():
             return []
         
+        # Fallback to character-based if tokenizer unavailable
+        if self.tokenizer is None:
+            return self._chunk_text_character_based(text, metadata)
+        
+        return self._chunk_text_token_based(text, metadata)
+    
+    def _chunk_text_token_based(self, text: str, metadata: Dict[str, Any] = None) -> List[Dict[str, Any]]:
```

Ruff (0.12.2)

`_chunk_text_token_based` is too complex (12 > 10)

(C901)